packages/design-system-react/src/components/button-primary/ButtonPrimary.test.tsx

```diff
@@ -172,4 +169,116 @@ describe('ButtonPrimary', () => {
     expect(link).toBeInTheDocument();
     expect(link).toHaveAttribute('href', '#');
   });
+
+  describe('loading state', () => {
```

Comment:
Adding more tests to ensure we still have 100% coverage after all of the new updates for loading state